### ISSUE TITLE
Adding Gaussian Noise into Inputs/Parameters/Labels

### ISSUE THREAD

**Issue by Demirrr** [dice-group/dice-embeddings#106]
# Adding Gaussian Noise into Inputs/Parameters
During training, we can add a gaussian noise to regularized models.

Previously, [Bishop](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/bishop-tikhonov-nc-95.pdf) show that Training with Noise is Equivalent to Tikhonov Regularization provided that a specific family of loss functions used.

Possible solutions:

1. Write a callback that add gaussian noise into embeddings at the end of each epoch
2. Write a callback (not quite sure how possible) that add gaussian noise into all embeddings found in a given batch.

Standard Evaluation:
1. Evaluate the link prediction performance of models w/o added gaussian noise

- [ ] Writing a callback that is a subclass of `AbstractCallback` adss gaussian noise to all `torch.nn.Module().parameters()` might be a viable way. @renzhonglu11 Noise can be sampled from a gaussian distribution N(0,std), where std can be determined by the user. The sampled scalar values are added into `torch.nn.Module().parameters()` See [torch.normal](https://pytorch.org/docs/stable/generated/torch.normal.html#torch-normal) for sampling from a gaussian distribution with mean and std.
- [ ] STD can be extracted from the callback name or we can find a better way of integrating callback related arguments

# Adding Gaussian Noise into Labels
Todo @Demirrr explain

**Comment by Demirrr** [comment]
Writing a callback that is a subclass of `AbstractCallback` adss gaussian noise to all `torch.nn.Module().parameters()` might be a viable way. @renzhonglu11

**Comment by Demirrr** [comment]
Noise can be sampled from a gaussian distribution N(0,std), where std can be determined by the user. The sampled scalar values are added into `torch.nn.Module().parameters()`
See [torch.normal](https://pytorch.org/docs/stable/generated/torch.normal.html#torch-normal) for sampling from a gaussian distribution with mean and std.

**Comment by renzhonglu11** [comment]
Dictionary-Like Key-Value Pairs of callbacks can be parsed now. Trainer of PL and TorchTrainer can now use the GN callback. But there are some issues with the other callbacks to pass to the PL trainer. May need to discuss this issue later.

**Comment by Demirrr** [comment]
https://github.com/dice-group/dice-embeddings/pull/113 pull fixed this issue. Thank you @renzhonglu11
